x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Analyzer.java

```diff
@@ -1243,6 +1248,25 @@ private Expression resolveConvertFunction(AbstractConvertFunction convert, List<
             return convert;
         }
 
+        private static Expression resolveFoldablesConvertFunction(FoldablesConvertFunction fcf) {
+            if (fcf.field() instanceof FieldAttribute fa && fa.field() instanceof InvalidMappedField imf) {
+                String unresolvedMessage = "argument of ["
+                    + fcf.sourceText()
+                    + "] must be a constant, received ["
+                    + Expressions.name(fa)
+                    + "]";
+                String types = imf.getTypesToIndices().keySet().stream().collect(Collectors.joining(","));
+                Expression ua = new UnsupportedAttribute(
+                    fa.source(),
+                    fa.name(),
+                    new UnsupportedEsField(imf.getName(), types),
+                    unresolvedMessage
+                );
+                return fcf.replaceChildren(Collections.singletonList(ua));
```

Comment:
bikeshed: we could probably also replace the whole convert function by an `UnresolvedAttribute` (with the custom message from above); probably irrelevant for the final verification, though.

Reply:
I'd prefer to keep the function name, not reverting it back to `UnresolvedAttribute`, as `ResolveFunctions` has don't its job, and it is correct. The conversion function's name has been resolved by `ResolveFunctions`, however its children haven't been resolved yet.